docs/src/main/sphinx/routines/examples.md

```diff
@@ -413,3 +413,112 @@ SELECT from_date_string('2023010123'); -- 2023-01-01 23:00:00.000 UTC (using the
 SELECT from_date_string(NULL); -- NULL (handles NULL string)
 SELECT from_date_string('abc'); -- NULL (not matched to any format)
 ```
+
+## Truncating long strings
+
+This example routine `strtrunc` truncates strings longer than 60 characters,
+leaving the first 30 and the last 25 characters, and cutting out extra
+characters in the middle.
+
+```sql
+FUNCTION strtrunc(input VARCHAR)
+RETURNS VARCHAR
+RETURN
+    CASE WHEN length(input) > 60
+    THEN substr(input, 1, 30) || ' ... ' || substr(input, length(input) - 25)
+    ELSE input
+    END;
+```
+
+The preceding declaration is very compact and consists of only one complex
+statement with a `CASE` expression and multiple function calls. It can therefore
+define the complete logic in the `RETURN` clause.
+
+The following statement shows the same capability within the function itself.
+Note the duplicate `RETURN` inside and outside the `CASE` statement and the
```

Comment:
I think this makes it not the best example. Maybe declaring a var and assigning it in case statements would look better, but honestly this alternative version seems unnecessary. The short version shows how even simple routines can be useful and improve readability of queries by hiding mundane details.

Reply:
I think this is crucially important to explain .. most users wont know about the CASE expression enough and look at CASE statement in routine docs instead .. and wonder why the difference. In fact that brings up a good point.. we should add some to both sections in the docs and link back and forth